raiden-dapp/src/services/raiden-service.ts

```diff
@@ -259,7 +259,17 @@ export default class RaidenService {
   async transfer(token: string, target: string, amount: BigNumber) {
     try {
       await this.raiden.getAvailability(target);
-      await this.raiden.transfer(token, target, amount);
+      const secretHash = await this.raiden.transfer(token, target, amount);
+
+      // Wait for transaction to be completed
+      await this.raiden.transfers$
```

Comment:
@andrevmatos it might make sense to have some kind of follow up ticket that implements timeout handling here, e.g. if the user declines transactions in MetaMask. What do you think?

Reply:
@nephix I would not suggest the timeout, or else it may give incomplete results. The transfer will ALWAYS complete eventually, in the worst case after `LockExpired` `Processed` (which may take several minutes, but the idea is still to give the proper feedback for the user here)

Reply:
If the user declines the first prompt on Metamask, the `transfer` promise will reject right alway. If they reject any of the following ones, it'll eventually be asked again (upon messages retries) until transfer expires or they accept it, so no exception case should be fatal or leave a transfer pending forever. Even closing the channel mid-transfer completes it.

Reply:
Thanks for the clarification @andrevmatos